src/lsm/compaction.zig

```diff
@@ -315,6 +330,26 @@ pub fn CompactionType(
             assert(compaction.io_pending == 0);
             assert(!compaction.merge_done);
 
+            // Move table optimization
```

Comment:
We can drop this comment since it's echoing the if condition mostly. Otherwise, we should provide more detail and especially motivation in the comment.

Also, comments always end with punctuation (`.` or `:`).